test/app.test.js

```diff
@@ -674,13 +674,7 @@ describe('app', function() {
   describe('app.model(ModelCtor, config)', function() {
     it('attaches the model to a datasource', function() {
       app.dataSource('db', { connector: 'memory' });
-      var TestModel = loopback.Model.extend('TestModel');
-      // TestModel was most likely already defined in a different test,
-      // thus TestModel.dataSource may be already set
-      delete TestModel.dataSource;
-
-      app.model(TestModel, { dataSource: 'db' });
-
+      app.model('TestModel', { dataSource: 'db' });
```

Comment:
Unless this test is running with a per-app registry, you need to keep the code deleting `TestModel.dataSource` around. Please add an assertion that `TestModel.dataSource` is not defined, that will show you whether you need to delete it first or not.

Note that if you remove the code and thus the dataSource will be already set, this test will become a tautology, i.e. it will always pass, regardless of the implementation.